src/legacy/core_plugins/embeddable_api/public/np_ready/public/lib/test_samples/embeddables/hello_world_container.tsx

```diff
@@ -34,32 +40,56 @@ export const HELLO_WORLD_CONTAINER = 'HELLO_WORLD_CONTAINER';
 type InheritedInput = {
   id: string;
   viewMode: ViewMode;
+  firstName: string;
   lastName: string;
 };
 
 interface HelloWorldContainerInput extends ContainerInput {
   lastName?: string;
 }
 
+interface HelloWorldContainerOptions {
+  getActions: GetActionsCompatibleWithTrigger;
+  getEmbeddableFactory: GetEmbeddableFactory;
+  getAllEmbeddableFactories: GetEmbeddableFactories;
+  overlays: CoreStart['overlays'];
+  notifications: CoreStart['notifications'];
+  inspector: InspectorStartContract;
+  SavedObjectFinder: React.ComponentType<any>;
+}
+
 export class HelloWorldContainer extends Container<InheritedInput, HelloWorldContainerInput> {
   public readonly type = HELLO_WORLD_CONTAINER;
 
-  constructor(input: ContainerInput, embeddableFactories: Map<string, EmbeddableFactory>) {
-    super(input, { embeddableLoaded: {} }, embeddableFactories);
+  constructor(
+    input: ContainerInput<{ firstName: string; lastName: string }>,
+    private readonly options: HelloWorldContainerOptions
+  ) {
+    super(input, { embeddableLoaded: {} }, options.getEmbeddableFactory);
   }
 
   public getInheritedInput(id: string) {
     return {
       id,
       viewMode: this.input.viewMode || ViewMode.EDIT,
+      firstName: '',
```

Comment:
why add this?  I don't think you should have too, and I think it makes less sense to have it here because the container doesn't ever change this value. `lastName` is passed in to this container so it could update it's children's `lastName` but it'll never change it's children's `firstName`.

Reply:
Removed.